tools/health/health_checker_pool.go

```diff
@@ -247,3 +247,103 @@ func (ncp *NodeHealthCheckerPool) WaitAllHashesEqual(height uint64) error {
 		}
 	}
 }
+
+func (ncp *NodeHealthCheckerPool) FindInconsistentHashesAtHeight(height uint64) map[string]sdk.Hash {
+	type CheckNodeResult struct {
+		Endpoint string
+		Hash     sdk.Hash
+	}
+
+	nodeCheckCh := make(chan CheckNodeResult)
+	nodeCount := len(ncp.nodeHealthCheckers)
+	for _, checker := range ncp.nodeHealthCheckers {
+		go func(checker *NodeHealthChecker) {
+			endpoint := checker.nodeInfo.Endpoint
+			identityKey := checker.nodeInfo.IdentityKey
+			hash, err := checker.BlockHash(height)
+			log.Printf("Node %s=%v has %s", endpoint, identityKey, hash)
+			if err != nil {
+				log.Printf("Error getting block hash from %s:%s\n", endpoint, err)
+				nodeCount--
+				return
+			}
+
+			nodeCheckCh <- CheckNodeResult{endpoint, hash}
+		}(checker)
+	}
+
+	hashes := make(map[sdk.Hash]int)
+	nodeHashResults := make(map[string]sdk.Hash)
+	for {
+		select {
+		case res := <-nodeCheckCh:
+			nodeHashResults[res.Endpoint] = res.Hash
+			hashes[res.Hash]++
+
+			hashCount := 0
+			for _, count := range hashes {
+				hashCount += count
+			}
+
+			if hashCount < nodeCount {
+				continue
+			}
+
+			if len(hashes) > 1 {
+				log.Printf("Block hashes differ (hash:count of returned): %v\n", hashes)
+				return nodeHashResults
+			}
+
+			log.Printf("All nodes got the same block hash at %d height\n", height)
+			return nil
+		}
+	}
+}
+
+func (ncp *NodeHealthCheckerPool) MaxHeight() uint64 {
```

Comment:
I think it's not a good idea to take the maxHeight of a node because if some nodes have the smaller height maybe there is a fork. We need to check that at height `n` all nodes have the same state and since this height start comparison